### PR TITLE
Allow hooks to selectively stop the CLI process

### PR DESCRIPTION
**Pull request by teobugslayer**
See https://github.com/NativeScript/nativescript-cli/issues/1337

### CODE REVIEW

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

File: services/hooks-service.ts [hunk 1]
```diff
@@ -99,7 +99,16 @@ export class HooksService implements IHooksService {
 					if (maybePromise) {
 						this.$logger.trace('Hook promises to signal completion');
 						let hookCompletion = new Future<void>();
-						maybePromise.then(() => hookCompletion.return(), (err: any) => hookCompletion.throw(err));
+						maybePromise.then(
+							() => hookCompletion.return(),
+							(err: any) => {
+								if (_.isBoolean(err.stopExecution) && err.stopExecution === false) {
```
Comment (rosen-vladimirov) [inline services/hooks-service.ts]: Isn't the,second part of the if enough (=== false)

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by Fatme** [comment]
:+1:

**Pull request merged**